File: src/Models/Supplier.java
package Models;

public class Supplier implements HasID{
    public int supplier_id;
    public String supplier_name;
    public String contact_email;
    public String phone;
    public String address;

    public Supplier(int supplier_id, String supplier_name, String contact_email, String phone, String address) {
        this.supplier_id = supplier_id;
        this.supplier_name = supplier_name;
        this.contact_email = contact_email;
        this.phone = phone;
        this.address = address;
    }

    public int getID() {
        return supplier_id;
    }

    public void setSupplier_id(int supplier_id) {
        this.supplier_id = supplier_id;
    }

    public String getSupplier_name() {
        return supplier_name;
    }

    public void setSupplier_name(String supplier_name) {
        this.supplier_name = supplier_name;
    }

    public String getContact_email() {
        return contact_email;
    }

    public void setContact_email(String contact_email) {
        this.contact_email = contact_email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Override
    public String toString() {
        return "Supplier{" +
                "supplier_id=" + supplier_id +
                ", supplier_name='" + supplier_name + '\'' +
                ", contact_email='" + contact_email + '\'' +
                ", phone='" + phone + '\'' +
                ", address='" + address + '\'' +
                '}';
    }
}

File: Python code/Entities/Supplier.py

class Supplier:
    def __init__(self, supplier_id, supplier_name, email, phone,address):
        self.supplier_id = supplier_id
        self.supplier_name=supplier_name
        self.address = address
        self.phone = phone
        self.email = email

    def get_supplier_id(self):
        return self.supplier_id

    def get_supplier_name(self):
        return self.supplier_name

    def get_address(self):
        return self.address

    def get_phone(self):
        return self.phone

    def get_email(self):
        return self.email

    def set_supplier_id(self, supplier_id):
        self.supplier_id = supplier_id

    def set_supplier_name(self, supplier_name):
        self.supplier_name = supplier_name

    def set_address(self, address):
        self.address = address

    def set_phone(self, phone):
        self.phone = phone

    def set_email(self, email):
        self.email = email

    def __str__(self):
        return f"Supplier ID: {self.supplier_id}, Name: {self.supplier_name}, Address: {self.address}, Phone: {self.phone}, Email: {self.email}"